Recipes/Telemetry/WebApiTelemetryExtensions.cs

```diff
@@ -33,23 +36,52 @@ internal static class WebApiTelemetryExtensions
                 telemetry.Succeeded = response.IsSuccessStatusCode;
                 telemetry.HttpStatusCode = response.StatusCode;
 
-                if (response.RequestMessage != null)
+                var request = response.RequestMessage;
+                if (request != null)
                 {
-                    telemetry.RequestUri = response.RequestMessage.RequestUri;
-
-                    var callerIpAddress = response.RequestMessage.CallerIpAddress();
-
-                    if (!string.IsNullOrWhiteSpace(callerIpAddress))
-                    {
-                        telemetry.IsIncomingRequest(true);
-                        telemetry.CallerIpAddress(callerIpAddress);
-                    }
+                    telemetry.RequestUri = request.RequestUri;
+                    telemetry.WithCallerIpAddressBasedOn(request);
+                    telemetry.WithOperationNameBasedOn(request);
                 }
             }
 
             return telemetry;
         }
 
+        private static void WithCallerIpAddressBasedOn(this Telemetry telemetry, HttpRequestMessage request)
+        {
+            var callerIpAddress = request.CallerIpAddress();
+
+            if (!string.IsNullOrWhiteSpace(callerIpAddress))
+            {
+                telemetry.IsIncomingRequest(true);
+                telemetry.CallerIpAddress(callerIpAddress);
+            }
+        }
+
+        private static void WithOperationNameBasedOn(this Telemetry telemetry, HttpRequestMessage request)
+        {
+            if (request.HasActionDescriptor())
+            {
+                telemetry.OperationName = request.GetActionDescriptor().ActionName;
+            }
+            else
+            {
+                var leftPart = request.RequestUri.GetLeftPart(UriPartial.Authority);
+                telemetry.OperationName = new Uri(leftPart).MakeRelativeUri(request.RequestUri).ToString();
+            }
+        }
+
+        private static bool HasActionDescriptor(this HttpRequestMessage request)
+        {
+            return request.GetActionDescriptor() != null;
+        }
+
+        public static Dictionary<string, string> ToDictionary(this HttpHeaders httpHeaders)
+        {

```

Comment:
If it's not used in this recipe, it probably doesn't belong here. It makes sense that it was extracted as part of refactoring, but it probably belongs with our telemetry code, not with Its.Log.